code/modules/client/preference/preferences.dm

```diff
@@ -1956,11 +2145,9 @@ var/global/list/special_role_times = list( //minimum age (in days) for accounts
 		H.g_headacc = g_headacc
 		H.b_headacc = b_headacc
 		H.ha_style = ha_style
-	if(character.species.bodyflags & HAS_MARKINGS)
-		character.r_markings = r_markings
-		character.g_markings = g_markings
-		character.b_markings = b_markings
-		character.m_style = m_style
+	if((character.species.bodyflags & HAS_HEAD_MARKINGS) || (character.species.bodyflags & HAS_BODY_MARKINGS) || (character.species.bodyflags & HAS_TAIL_MARKINGS))
```

Comment:
You should keep `HAS_MARKINGS` as a combo of these 3 flags, instead of checking each individually.